File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/infra/network/RetrofitInjector.java
package com.savvasdalkitsis.gameframe.injector.infra.network;

import android.support.annotation.Nullable;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

import static com.savvasdalkitsis.gameframe.injector.infra.network.InterceptorInjector.ipBaseHostInterceptor;
import static com.savvasdalkitsis.gameframe.injector.infra.network.OkHttpClientInjector.okHttpClient;

public class RetrofitInjector {

    @Nullable
    public static Retrofit retrofit() {
        return new Retrofit.Builder()
                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create())
                .baseUrl("http://nothing")
                .client(okHttpClient(3)
                        .addInterceptor(ipBaseHostInterceptor())
                        .build())
                .build();
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/infra/network/RetrofitInjector.kt
package com.savvasdalkitsis.gameframe.injector.infra.network

import com.savvasdalkitsis.gameframe.injector.infra.network.InterceptorInjector.ipBaseHostInterceptor
import com.savvasdalkitsis.gameframe.injector.infra.network.OkHttpClientInjector.okHttpClient
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitInjector {

    fun retrofit(): Retrofit = Retrofit.Builder()
            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
            .addConverterFactory(GsonConverterFactory.create())
            .baseUrl("http://nothing")
            .client(okHttpClient(3)
                    .addInterceptor(ipBaseHostInterceptor())
                    .build())
            .build()
}
